server_test.go

```diff
@@ -141,6 +145,81 @@ var _ = Describe("Server", func() {
 			})
 		})
 	})
+
+	Describe("prompts", func() {
+		BeforeEach(func() {
+			stdin.WriteString(`{"jsonrpc":"2.0","id":1,"method":"initialize","params":{"protocolVersion":"2024-11-05","capabilities":{"roots":{"listChanged":true},"sampling":{}},"clientInfo":{"name":"ExampleClient","version":"1.0.0"}}}`)
+			Eventually(session.Out).Should(gbytes.Say("ExampleServer"))
+			stdin.WriteString(`{"jsonrpc":"2.0","method":"initialized"}`)
+			Eventually(session.Out).Should(gbytes.Say("\n"))
+		})
+
+		Context("when the client requests the list of prompts", func() {
+			It("responds", func() {
+				stdin.WriteString(`{"jsonrpc":"2.0","id":2,"method":"prompts/list"}`)
+				Eventually(session.Out).Should(gbytes.Say("prompts"))
+				Expect(lastResponse(session.Out.Contents())).To(MatchJSON(`{"jsonrpc":"2.0","id":2,"result":{"prompts":[{"name":"example","description":"An example prompt template","arguments":[{"name":"text","description":"Text to process","required":true}]}]}}`))
+			})
+		})
+
+		Context("when the client requests the list of prompts with an invalid cursor", func() {
+			It("responds with a protocol error", func() {
+				stdin.WriteString(`{"jsonrpc":"2.0","id":2,"method":"prompts/list","params":{"cursor":"invalid-cursor"}}`)
+				Eventually(session.Out).Should(gbytes.Say("\n"))
+				Expect(lastResponse(session.Out.Contents())).To(MatchJSON(`{"jsonrpc":"2.0","error":{"code":-32602,"message":"Invalid params"},"id":2}`))
+			})
+		})
+
+		Context("when the client gets a prompt", func() {
+			It("processes the prompt with the provided arguments", func() {
+				stdin.WriteString(`{"jsonrpc":"2.0","id":2,"method":"prompts/get","params":{"name":"example","arguments":{"text":"test input"}}}`)
+				Eventually(session.Out).Should(gbytes.Say("\n"))
+
+				responses := allResponses(session.Out.Contents())
+				Expect(responses).To(HaveLen(4)) // First two are initialization, last two are the prompt response
+				Expect(responses[len(responses)-2]).To(MatchJSON(`{"jsonrpc":"2.0","method":"test/notification","params":{"message":"Processing text"}}`))
+				Expect(responses[len(responses)-1]).To(MatchJSON(`{"jsonrpc":"2.0","id":2,"result":{"messages":[{"role":"assistant","content":{"type":"text","text":"Processed: test input"}}]}}`))
+			})
+
+			// Missing arguments error
```

Comment:
The comments for these new test examples don't seem necessary as they match the context descriptions?